src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs

```diff
@@ -1255,24 +1255,32 @@ private void DecodeInterpolatedStringHandlerArgumentAttribute(ref DecodeWellKnow
                 diagnostics.Add(ErrorCode.WRN_InterpolatedStringHandlerArgumentAttributeIgnoredOnLambdaParameters, arguments.AttributeSyntaxOpt.Location);
             }
 
+            if (ContainingSymbol is SynthesizedExtensionMarker { Parameters: [var param, ..] } && (object)param == this)
+            {
+                // Interpolated string handler arguments are not allowed in this context.
+                diagnostics.Add(ErrorCode.ERR_InterpolatedStringHandlerArgumentDisallowed, arguments.AttributeSyntaxOpt.Location);
+                setInterpolatedStringHandlerAttributeError(ref arguments);
+                return;
+            }
+
             TypedConstant constructorArgument = arguments.Attribute.CommonConstructorArguments[0];
 
             ImmutableArray<ParameterSymbol> containingSymbolParameters = ContainingSymbol.GetParameters();
+            // If we're analyzing the extension marker method itself, then we're not actually in an extension context, even though our containing type
+            // is the extension marker type.
```

Comment:
>// is the extension marker type. [](http://example.com/codeflow?start=12&length=32)

There is still problem with terminology. There is no "extension marker type" in source. In source, there is an extension block and a type representing it. However, it looks like we can remove the comment and the next line since it should have no effect now.